### PR TITLE
Add jupyterlab-plotly-extension

### PR DESCRIPTION
**Pull request by dbast**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dbast** [comment]
Built+installed the package locally and it works here.

Pinging @blink1073 @uday1889 and @conda-forge/jupyterlab for review.

Open question: should the package be named `jupyterlab_plotly-extension` or `jupyterlab-plotly-extension` (underscrore vs. hyphen, there also already is jupyterlab-git)?

(The `python >=3.5` is there to ensure the noarch build is not done with python 2.7, where no jupyterlab is available.)

@blink1073 and @uday1889 are you fine co-maintaining this?

Thanks

**Comment by blink1073** [comment]
I'd say all hyphens, happy to be a maintainer for this.  Thanks!

**Comment by dbast** [comment]
I am wondering if this is really the best way to provide frontend only jupyterlab extensions:
* For some extensions like jupyterlab-git, the package is a combination of a server extension written in python + frontend code in nodejs. Providing this as conda package makes the installation easier.
* other extensions, like this one, are frontend only (contain no python part) and could instead also be installed via the Jupyterlab Extension manager only... 

In both cases Jupyterlab requires to do a nodejs extension rebuild, which would be nice to have it already in a pre-build package, but afaik currently by design can not be done during the conda package build.

The npm stats show that many frontend extensions are installed from npm directly.

On the other side having frontend only extensions all with a version specified as part of an environment.yaml, is still appealing, even though the nodejs rebuild step+time is remaining.

This PR could also serve as an example to also bring some (not all, as some are already integrated into default Juputerlab) of the other frontend extensions from https://github.com/jupyterlab/jupyter-renderers to conda-forge.

Another interesting frontend only extension is https://github.com/bokeh/jupyterlab_bokeh

Whereas this is has a server and frontend side, which makes packaging more interesting https://github.com/dask/dask-labextension

tl;dr frontend only JLab extensions as package: Yes/No? What would be the conda-forge/JLab canonical way?

@blink1073 @conda-forge/jupyterlab @conda-forge/staged-recipes Any opinion?

Thanks

**Comment by blink1073** [comment]
I like the idea of being able to use environment.yaml for frontend extensions

**Comment by dbast** [comment]
To connect the dots, also linking to https://github.com/jupyterlab/jupyterlab/issues/2065

File: recipes/jupyterlab-plotly-extension/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  summary: A JupyterLab extension for rendering Plotly charts
+  dev_url: https://github.com/jupyterlab/jupyter-renderers
+
+extra:
+  recipe-maintainers:
+    - dbast
+    - blink1073
+    - uday1889
```
Comment (scopatz) [inline recipes/jupyterlab-plotly-extension/meta.yaml]: @uday1889 - are you OK with being a maintainer here?

Comment (uday1889) [reply recipes/jupyterlab-plotly-extension/meta.yaml]: @scopatz: Sure. That's fine with me.

**Comment by scopatz** [comment]
Merging this, but please bring up the issue of how to do this on https://github.com/conda-forge/conda-forge.github.io

**Comment by dbast** [comment]
@scopatz Thanks. Also opened https://github.com/conda-forge/conda-forge.github.io/issues/761 ...
